src/visitable.cpp

```diff
@@ -717,13 +731,13 @@ static long charges_of_internal( const T& self, const itype_id& id, long limit )
 }
 
 template <typename T>
-long visitable<T>::charges_of( const std::string &what, int limit ) const
+long visitable<T>::charges_of( const std::string &what, long limit ) const
 {
     return charges_of_internal( *this, what, limit );
 }
 
 template <>
-long visitable<inventory>::charges_of( const std::string &what, int limit ) const
+long visitable<inventory>::charges_of( const std::string &what, long limit ) const
```

Comment:
There is confusion over whether you're using ints or longs; this code uses longs, but sum_no_wrap uses ints.